src/lib.rs

```diff
@@ -355,8 +388,8 @@ impl<D: tb::Bool, H: tb::Bool, T: tb::Bool> InvoiceBuilder<D, H, T> {
 	}
 
 	/// Sets the expiry time in seconds.
-	pub fn expiry_time_seconds(mut self, expiry_seconds: u64) -> Self {
-		self.tagged_fields.push(TaggedField::ExpiryTime(ExpiryTime {seconds: expiry_seconds}));
+	pub fn expiry_time(mut self, expiry_time: Duration) -> Self {
```

Comment:
This should have a doc that indicates it will panic if expiry_time is greater than a year (well, currently MAX_UNIX_TIMESTAMP - a year, but I think thats a bug?), or probably add the doc and then instead of unwrap() store the Err in self.error?

Reply:
That's a bug, I added proper error handling.